[2024-11-09T12:24:42.051+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2024-11-09T12:24:42.072+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: mongodb_to_postgres.fetch_data manual__2024-11-09T12:24:38.504802+00:00 [queued]>
[2024-11-09T12:24:42.083+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: mongodb_to_postgres.fetch_data manual__2024-11-09T12:24:38.504802+00:00 [queued]>
[2024-11-09T12:24:42.083+0000] {taskinstance.py:2866} INFO - Starting attempt 1 of 1
[2024-11-09T12:24:42.101+0000] {taskinstance.py:2889} INFO - Executing <Task(PythonOperator): fetch_data> on 2024-11-09 12:24:38.504802+00:00
[2024-11-09T12:24:42.112+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70: DeprecationWarning: This process (pid=1117) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2024-11-09T12:24:42.113+0000] {standard_task_runner.py:72} INFO - Started process 1119 to run task
[2024-11-09T12:24:42.114+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'mongodb_to_postgres', 'fetch_data', 'manual__2024-11-09T12:24:38.504802+00:00', '--job-id', '45', '--raw', '--subdir', 'DAGS_FOLDER/et_pipeline.py', '--cfg-path', '/tmp/tmptiqh5502']
[2024-11-09T12:24:42.116+0000] {standard_task_runner.py:105} INFO - Job 45: Subtask fetch_data
[2024-11-09T12:24:42.170+0000] {task_command.py:467} INFO - Running <TaskInstance: mongodb_to_postgres.fetch_data manual__2024-11-09T12:24:38.504802+00:00 [running]> on host 501997186f25
[2024-11-09T12:24:42.558+0000] {taskinstance.py:3132} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='mongodb_to_postgres' AIRFLOW_CTX_TASK_ID='fetch_data' AIRFLOW_CTX_EXECUTION_DATE='2024-11-09T12:24:38.504802+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-11-09T12:24:38.504802+00:00'
[2024-11-09T12:24:42.559+0000] {logging_mixin.py:190} INFO - Task instance is in running state
[2024-11-09T12:24:42.559+0000] {logging_mixin.py:190} INFO -  Previous state of the Task instance: queued
[2024-11-09T12:24:42.560+0000] {logging_mixin.py:190} INFO - Current task name:fetch_data state:running start_date:2024-11-09 12:24:42.072667+00:00
[2024-11-09T12:24:42.560+0000] {logging_mixin.py:190} INFO - Dag name:mongodb_to_postgres and current dag run status:running
[2024-11-09T12:24:42.561+0000] {taskinstance.py:731} INFO - ::endgroup::
[2024-11-09T12:24:42.575+0000] {base.py:84} INFO - Retrieving connection 'mongo_default'
[2024-11-09T12:24:44.701+0000] {python.py:240} INFO - Done. Returned value was: [{'_id': ObjectId('672f31891e17ec3a269dc22e'), 'journal': 'Published in: IEEE Access ( Volume: 11)', 'indexation': 'IEEE', 'publication': 'Publisher: IEEE', 'doi': 'DOI: 10.1109/ACCESS.2023.3325325', 'titre': 'A Systematic Framework for Addressing Critical Challenges in Adopting DevOps Culture in Software Development: A PLS-SEM Perspective', 'chercheurs': 'Kausar-Nasreen Khattak; Faiza Qayyum; Syed Shehryar Ali Naqvi; Asif Mehmood; Jungsuk Kim', 'laboratoires': 'Non disponible', 'abstract': 'DevOps is the bundle of applications and culture activity that focuses to separate the challenges between evaluation and performance associates to boost participation and communication. Several companies clutch DevOps fundamentals due to tremendous perspectives, like small tempo for production, rise accuracy, and durability. In contempt of the global assumption of DevOps and its framework, there is an absence of interpretation studies and essential ideas, implementation, and issues related to executing DevOps procedures. The major objective of this research study is to investigate and study issues affiliated with the DevOps culture (DVC) and applications. The challenges that are challenging in adopting DevOps culture in software development (SD) are inspected by conducting the systematic literature review (SLR). Furthermore, it explains the working of DevOps in industries, gives full clarification of DevOps, and probes the cultural issues faced by the industries while executing DevOps. This study discloses ten barriers that should be declamation in accepting DevOps culture. In accordance with the discoveries, these ten barriers contemplate during the execution of DevOps culture i.e. lack of collaboration and coordination, Lack of skill and knowledge, complicated infrastructure, Lack of management, Lack of DevOps approach, and trust confidence problems.', 'ISSN': {'Print ISSN': 'Non disponible', 'Electronic ISSN': '2169-3536', 'CD': 'Non disponible'}, 'quartile': 'Journal pas indexé Scopus', 'pays': None}, {'_id': ObjectId('672f3223911cff78ae03550d'), 'journal': 'Published in: IEEE Access ( Volume: 11)', 'indexation': 'IEEE', 'publication': 'Publisher: IEEE', 'doi': 'DOI: 10.1109/ACCESS.2023.3325325', 'titre': 'A Systematic Framework for Addressing Critical Challenges in Adopting DevOps Culture in Software Development: A PLS-SEM Perspective', 'chercheurs': 'Kausar-Nasreen Khattak; Faiza Qayyum; Syed Shehryar Ali Naqvi; Asif Mehmood; Jungsuk Kim', 'laboratoires': 'Non disponible', 'abstract': 'DevOps is the bundle of applications and culture activity that focuses to separate the challenges between evaluation and performance associates to boost participation and communication. Several companies clutch DevOps fundamentals due to tremendous perspectives, like small tempo for production, rise accuracy, and durability. In contempt of the global assumption of DevOps and its framework, there is an absence of interpretation studies and essential ideas, implementation, and issues related to executing DevOps procedures. The major objective of this research study is to investigate and study issues affiliated with the DevOps culture (DVC) and applications. The challenges that are challenging in adopting DevOps culture in software development (SD) are inspected by conducting the systematic literature review (SLR). Furthermore, it explains the working of DevOps in industries, gives full clarification of DevOps, and probes the cultural issues faced by the industries while executing DevOps. This study discloses ten barriers that should be declamation in accepting DevOps culture. In accordance with the discoveries, these ten barriers contemplate during the execution of DevOps culture i.e. lack of collaboration and coordination, Lack of skill and knowledge, complicated infrastructure, Lack of management, Lack of DevOps approach, and trust confidence problems.', 'ISSN': {'Print ISSN': 'Non disponible', 'Electronic ISSN': '2169-3536', 'CD': 'Non disponible'}, 'quartile': 'Journal pas indexé Scopus', 'pays': None}, {'_id': ObjectId('672f329c49067b356a34e56f'), 'journal': 'Published in: IEEE Access ( Volume: 11)', 'indexation': 'IEEE', 'publication': 'Publisher: IEEE', 'doi': 'DOI: 10.1109/ACCESS.2023.3325325', 'titre': 'A Systematic Framework for Addressing Critical Challenges in Adopting DevOps Culture in Software Development: A PLS-SEM Perspective', 'chercheurs': 'Kausar-Nasreen Khattak; Faiza Qayyum; Syed Shehryar Ali Naqvi; Asif Mehmood; Jungsuk Kim', 'laboratoires': 'Non disponible', 'abstract': 'DevOps is the bundle of applications and culture activity that focuses to separate the challenges between evaluation and performance associates to boost participation and communication. Several companies clutch DevOps fundamentals due to tremendous perspectives, like small tempo for production, rise accuracy, and durability. In contempt of the global assumption of DevOps and its framework, there is an absence of interpretation studies and essential ideas, implementation, and issues related to executing DevOps procedures. The major objective of this research study is to investigate and study issues affiliated with the DevOps culture (DVC) and applications. The challenges that are challenging in adopting DevOps culture in software development (SD) are inspected by conducting the systematic literature review (SLR). Furthermore, it explains the working of DevOps in industries, gives full clarification of DevOps, and probes the cultural issues faced by the industries while executing DevOps. This study discloses ten barriers that should be declamation in accepting DevOps culture. In accordance with the discoveries, these ten barriers contemplate during the execution of DevOps culture i.e. lack of collaboration and coordination, Lack of skill and knowledge, complicated infrastructure, Lack of management, Lack of DevOps approach, and trust confidence problems.', 'ISSN': {'Print ISSN': 'Non disponible', 'Electronic ISSN': '2169-3536', 'CD': 'Non disponible'}, 'quartile': 'Journal pas indexé Scopus', 'pays': None}, {'_id': ObjectId('672f33a794e00f15d188bec8'), 'journal': 'Published in: IEEE Access ( Volume: 11)', 'indexation': 'IEEE', 'publication': 'Publisher: IEEE', 'doi': 'DOI: 10.1109/ACCESS.2023.3325325', 'titre': 'A Systematic Framework for Addressing Critical Challenges in Adopting DevOps Culture in Software Development: A PLS-SEM Perspective', 'chercheurs': 'Kausar-Nasreen Khattak; Faiza Qayyum; Syed Shehryar Ali Naqvi; Asif Mehmood; Jungsuk Kim', 'laboratoires': 'Kausar-Nasreen Khattak; Faiza Qayyum; Syed Shehryar Ali Naqvi; Asif Mehmood; Jungsuk Kim', 'abstract': 'DevOps is the bundle of applications and culture activity that focuses to separate the challenges between evaluation and performance associates to boost participation and communication. Several companies clutch DevOps fundamentals due to tremendous perspectives, like small tempo for production, rise accuracy, and durability. In contempt of the global assumption of DevOps and its framework, there is an absence of interpretation studies and essential ideas, implementation, and issues related to executing DevOps procedures. The major objective of this research study is to investigate and study issues affiliated with the DevOps culture (DVC) and applications. The challenges that are challenging in adopting DevOps culture in software development (SD) are inspected by conducting the systematic literature review (SLR). Furthermore, it explains the working of DevOps in industries, gives full clarification of DevOps, and probes the cultural issues faced by the industries while executing DevOps. This study discloses ten barriers that should be declamation in accepting DevOps culture. In accordance with the discoveries, these ten barriers contemplate during the execution of DevOps culture i.e. lack of collaboration and coordination, Lack of skill and knowledge, complicated infrastructure, Lack of management, Lack of DevOps approach, and trust confidence problems.', 'ISSN': {'Print ISSN': 'Non disponible', 'Electronic ISSN': '2169-3536', 'CD': 'Non disponible'}, 'quartile': 'Journal pas indexé Scopus', 'pays': None}]
[2024-11-09T12:24:44.710+0000] {xcom.py:690} ERROR - Object of type ObjectId is not JSON serializable. If you are using pickle instead of JSON for XCom, then you need to enable pickle support for XCom in your *** config or make sure to decorate your object with attr.
[2024-11-09T12:24:44.712+0000] {taskinstance.py:3311} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/json.py", line 92, in default
    return serialize(o)
           ^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/serialization/serde.py", line 190, in serialize
    raise TypeError(f"cannot serialize object of type {cls}")
TypeError: cannot serialize object of type <class 'bson.objectid.ObjectId'>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 789, in _execute_task
    task_instance.xcom_push(key=XCOM_RETURN_KEY, value=xcom_value, session=session_or_null)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 94, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3639, in xcom_push
    XCom.set(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/api_internal/internal_api_call.py", line 166, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 94, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/xcom.py", line 249, in set
    value = cls.serialize_value(
            ^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/xcom.py", line 688, in serialize_value
    return json.dumps(value, cls=XComEncoder).encode("UTF-8")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/json/__init__.py", line 238, in dumps
    **kw).encode(obj)
          ^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/json.py", line 105, in encode
    return super().encode(o)
           ^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/json/encoder.py", line 200, in encode
    chunks = self.iterencode(o, _one_shot=True)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/json/encoder.py", line 258, in iterencode
    return _iterencode(o, 0)
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/json.py", line 94, in default
    return super().default(o)
           ^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/json/encoder.py", line 180, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type ObjectId is not JSON serializable
[2024-11-09T12:24:44.723+0000] {logging_mixin.py:190} INFO - Task instance in failure state
[2024-11-09T12:24:44.724+0000] {logging_mixin.py:190} INFO - Task start:2024-11-09 12:24:42.072667+00:00 end:2024-11-09 12:24:44.723297+00:00 duration:2.65063
[2024-11-09T12:24:44.724+0000] {logging_mixin.py:190} INFO - Task:<Task(PythonOperator): fetch_data> dag:<DAG: mongodb_to_postgres> dagrun:<DagRun mongodb_to_postgres @ 2024-11-09 12:24:38.504802+00:00: manual__2024-11-09T12:24:38.504802+00:00, state:running, queued_at: 2024-11-09 12:24:38.567326+00:00. externally triggered: True>
[2024-11-09T12:24:44.724+0000] {logging_mixin.py:190} INFO - Failure caused by Object of type ObjectId is not JSON serializable
[2024-11-09T12:24:44.724+0000] {taskinstance.py:1225} INFO - Marking task as FAILED. dag_id=mongodb_to_postgres, task_id=fetch_data, run_id=manual__2024-11-09T12:24:38.504802+00:00, execution_date=20241109T122438, start_date=20241109T122442, end_date=20241109T122444
[2024-11-09T12:24:44.740+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2024-11-09T12:24:44.740+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 45 for task fetch_data (Object of type ObjectId is not JSON serializable; 1119)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/json.py", line 92, in default
    return serialize(o)
           ^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/serialization/serde.py", line 190, in serialize
    raise TypeError(f"cannot serialize object of type {cls}")
TypeError: cannot serialize object of type <class 'bson.objectid.ObjectId'>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3005, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3159, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3183, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 789, in _execute_task
    task_instance.xcom_push(key=XCOM_RETURN_KEY, value=xcom_value, session=session_or_null)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 94, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3639, in xcom_push
    XCom.set(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/api_internal/internal_api_call.py", line 166, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 94, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/xcom.py", line 249, in set
    value = cls.serialize_value(
            ^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/xcom.py", line 688, in serialize_value
    return json.dumps(value, cls=XComEncoder).encode("UTF-8")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/json/__init__.py", line 238, in dumps
    **kw).encode(obj)
          ^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/json.py", line 105, in encode
    return super().encode(o)
           ^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/json/encoder.py", line 200, in encode
    chunks = self.iterencode(o, _one_shot=True)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/json/encoder.py", line 258, in iterencode
    return _iterencode(o, 0)
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/json.py", line 94, in default
    return super().default(o)
           ^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/json/encoder.py", line 180, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type ObjectId is not JSON serializable
[2024-11-09T12:24:44.779+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2024-11-09T12:24:44.806+0000] {taskinstance.py:3895} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-11-09T12:24:44.813+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
